docs/data/material/migration/migrating-to-v6/migrating-to-pigment-css.md

```diff
@@ -0,0 +1,484 @@
+
+Before going through this guide, make sure you have Material UI v6 installed. If you have not yet migrated to Material UI v6, follow the [migration guide](/material-ui/migration/migrating-to-v6/) first.
+
+## Supported frameworks
+
+Pigment CSS can be used with one of the following frameworks:
+
+- [Next.js](https://nextjs.org/) with Webpack 5 (Turbo is not supported yet)
+- [Vite](https://vitejs.dev/)
+
+## Installation
+
+First, install the dependency:
+
+<codeblock storageKey="package-manager">
+
+```bash npm
+npm install @pigment-css/react
+```
+
+```bash yarn
+yarn add @pigment-css/react
+```
+
+```bash pnpm
+pnpm add @pigment-css/react
+```
+
+</codeblock>
+
+Then, follow the instructions based on your framework:
+
+### Next.js
+
+Install the Next.js plugin as a dev dependency:
+
+<codeblock storageKey="package-manager">
+
+```bash npm
+npm install --save-dev @pigment-css/nextjs-plugin
+```
+
+```bash yarn
+yarn add -D @pigment-css/nextjs-plugin
+```
+
+```bash pnpm
+pnpm add -D @pigment-css/nextjs-plugin
+```
+
+</codeblock>
+
+Next, open Next.js config file and add the plugin:
+
+<codeblock>
+
+```js next.config.mjs
+import { withPigment } from '@pigment-css/nextjs-plugin';
+
+const nextConfig = {
+  // ...Your nextjs config.
+};
+
+export default withPigment(nextConfig);
+```
+
+```js next.config.js
+const { withPigment } = require('@pigment-css/nextjs-plugin');
+
+const nextConfig = {
+  // ...Your nextjs config.
+};
+
+module.exports = withPigment(nextConfig);
+```
+
+</codeblock>
+
+Finally, import the stylesheet to your application.
+
+```js title="app/layout.(js|tsx)"
+// ...other imports
+import '@pigment-css/react/styles.css';
+```
+
+### Vite
+
+Install the Vite plugin as a dev dependency:
+
+<codeblock storageKey="package-manager">
+
+```bash npm
+npm install --save-dev @pigment-css/vite-plugin
+```
+
+```bash yarn
+yarn add -D @pigment-css/vite-plugin
+```
+
+```bash pnpm
+pnpm add -D @pigment-css/vite-plugin
+```
+
+</codeblock>
+
+Next, open vite config file, usually named `vite.config.js`, and add the plugin:
+
+```js
+import { defineConfig } from 'vite';
+import { pigment } from '@pigment-css/vite-plugin';
+
+export default defineConfig({
+  plugins: [
+    pigment(),
+    // ... Your other plugins.
+  ],
+});
+```
+
+Finally, add the Pigment CSS stylesheet to your application.
+
+```js title="src/main.(js|tsx)"
+// ...other imports
+import '@pigment-css/react/styles.css';
+```
+
+## Configuring the theme
+
+Integrating Pigment CSS with Material UI requires you to configure the theme to the plugin.
+The theme must be created from `extendTheme` utility from `@mui/material` package.
+
+<codeblock>
+
+```js next.config.mjs
+import { withPigment } from '@pigment-css/nextjs-plugin';
+import { extendTheme, stringifyTheme } from '@mui/material';
+
+const customTheme = extendTheme({
+  // ...Your Material UI theme.
+});
+customTheme.toRuntimeSource = stringifyTheme;
```

Comment:
Would it makes sense for the `extendTheme` from `@mui/material` to already have the `toRuntimeSource` baked in?

Reply:
I don't think so, it's too specific for Pigment. Let me find a better way with '@mui/material-pigment-css'.